src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -107,10 +106,95 @@ private static Schema inferSchema(final String fieldPath, final BsonValue bsonVa
       case MAX_KEY:
       case UNDEFINED:
       default:
-        return DEFAULT_INFER_SCHEMA_TYPE;
+        return Schema.OPTIONAL_STRING_SCHEMA;
     }
   }
 
+  private static Schema combineArrayValueSchema(
+      final Schema firstSchema, final Schema secondSchema) {
+    if (isSentinel(firstSchema)) {
+      return secondSchema;
+    } else if (isSentinel(secondSchema)) {
```

Comment:
Looks like this condition was actually required.  How come?

Reply:
Its just a logic shortcut - it will fall through if not added.